job_dsl/rpc_gating_merge_trigger.groovy

```diff
@@ -0,0 +1,78 @@
+// Can't use env.FOO = {FOO} to transfer JJB vars to groovy
+// as this file won't be templated by JJB.
+// Alternative is to use parameters with JJB vars as the defaults.
+library "rpc-gating@${RPC_GATING_BRANCH}"
+common.globalWraps(){
+
+  // We do not want to trigger any of these
+  // for rpc-gating forks or branches other
+  // than 'master'. So if this is triggered
+  // by anything else, end now.
+  // TODO(odyssey4me):
+  // Remove once RE-1645 is resolved.
+  if (env.RPC_GATING_BRANCH != "master") {
+    print "Triggered by branch ${RPC_GATING_BRANCH} instead of master. Skipping triggers."
+    currentBuild.result = 'SUCCESS'
+    return
+  }
+
+  stage("Build-Gating-Venv") {
+    build(
+      job: "Build-Gating-Venv",
+      wait: true,
+      parameters: [
+        [
+          $class: 'StringParameterValue',
+          name: 'RPC_GATING_BRANCH',
+          value: env.RPC_GATING_BRANCH
+        ]
+      ]
+    ) // build
+  } // stage
+
+  stage("Jenkins-Job-Builder") {
+    build(
+      job: "Jenkins-Job-Builder",
+      wait: true,
+      parameters: [
+        [
+          $class: 'StringParameterValue',
+          name: 'RPC_GATING_BRANCH',
+          value: env.RPC_GATING_BRANCH
+        ]
+      ]
+    ) // build
+  } // stage
+
+  // To prevent having to continually update a list of jobs to run
+  // in parallel once the two blocking jobs are complete, we match
+  // the jobs against a regex and execute everything that matches
+  // it in parallel.
+  jobNames = common.getMatchingJobs("(?:Build-Summary-Docker-Build|Setup-.*)")
```

Comment:
What's the meaning of `?:`? Also, this will match `Setup-`. Should we lock that down a bit tighter?

Reply:
``?:`` makes the () a non-capturing group, making just a small optimisation as we don't need to use a capture group anywhere. https://www.regular-expressions.info/brackets.html

Reply:
By locking it down tighter, what exactly do you mean? The only jobs we have that I can see prefixed by 'Setup-' are the jobs which setup infrastructure for rpc-gating. We should probably making our naming conventions better, but this handles what we have.

Reply:
I wonder if a list of jobs would be easier. It would only need modifying when a job is added or removed. There doesn't appear to be a clear discriminator with the names so I wouldn't be surprised if merge jobs end up getting added that don't use this.